tests/org.eclipse.jface.tests/src/org/eclipse/jface/tests/images/FileImageDescriptorTest.java

```diff
@@ -174,4 +176,22 @@ public void testGetxPath150() {
 		assertEquals(Math.round(imageData.height * 1.5), imageDataZoomed.height);
 	}
 
+	public void testImageFileNameProviderGetxPath() {
+		ImageDescriptor descriptor = ImageDescriptor.createFromFile(FileImageDescriptorTest.class,
+				"/icons/imagetests/rectangular-57x16.png");
+		assertTrue("FileImageDescriptor does not implement IAdaptable", descriptor instanceof IAdaptable);
+		IAdaptable adaptable = (IAdaptable) descriptor;
+		ImageFileNameProvider fileNameProvider = adaptable.getAdapter(ImageFileNameProvider.class);
```

Comment:
Usually the pattern is to use:

Adapters.getAdapter

instead of checking for adaptable directly.

Reply:
I agree that this is what should be done in production code, but this unit test particularly wants to tests the IAdaptable inheritance.

Reply:
Does it really matter "how" it is adapted? I think the important part is *that* it could be adapted....

Reply:
Well, it does not matter to me, certainly. My test was constructed based on the use case in [MenuHelper.java#L491](https://github.com/eclipse-platform/eclipse.platform.ui/blob/849e023520392865995477474ef731f5302edbda/bundles/org.eclipse.ui.workbench/Eclipse%20UI/org/eclipse/ui/internal/menus/MenuHelper.java#L491) where also such a concrete tests exists. I have no strong feelings about this aspect. If replacing the concrete test for IAdaptable inheritance an its usage by a call to `org.eclipse.core.runtime.Adapters.adapt(Object sourceObject, Class<T> adapter)` or `org.eclipse.core.runtime.Adapters.of(Object sourceObject, Class<T> adapter)` would increase consensus, let me know.